VoodooRMI/Functions/F03.cpp

```diff
@@ -170,18 +170,21 @@ void F03::handlePacketGated(u8 packet)
     
     // The highest dx/dy is lowered by subtracting by trackstickDeadzone.
     // This however does allows values below the deadzone value to still be sent, preserving control in the lower end
-    dx = abs(dx) > trackstickDeadzone ? (abs(dx) - trackstickDeadzone) * signum(dx) : 0;
-    dx = abs(dy) > trackstickDeadzone ? (abs(dy) - trackstickDeadzone) * signum(dy) : 0;
-    
-    // Must multiply first then divide so we don't multiply by zero
-    if(buttons & 0x04 && (dx || dy)) {
-        buttonDevice->updateScrollwheel((-dy * trackstickScrollYMult) / DEFAULT_MULT,
-                                        (-dx * trackstickScrollXMult) / DEFAULT_MULT,
-                                        0);
-    } else {
-        buttonDevice->updateRelativePointer((dx * trackstickMult) / DEFAULT_MULT,
-                                            (dy * trackstickMult) / DEFAULT_MULT,
-                                            buttons);
+    
+    dx -= signum(dx) * min(abs(dx), trackstickDeadzone);
+    dy -= signum(dy) * min(abs(dy), trackstickDeadzone);
+    
+    if (dx && dy) {
+        // Must multiply first then divide so we don't multiply by zero
+        if(buttons & 0x04) {
+            buttonDevice->updateScrollwheel(-dy / DEFAULT_MULT * trackstickScrollYMult,
```

Comment:
Why divide first then multiply?
The order was very specific in the first place as dividng 4/5, 2/5, etc would give 0. 24/5 would be 4, etc. You lose a lot of resolution by dividing first, as you now make dx/dy *have* to be a factor of five (9/5 = 1, 1 * 5 = 5).

Reply:
I decided to divide first since multiplying first caused overflow, but I forgot about that. Too much Python is harmful. 
Now I did type casting toward 64-bit integer to prevent overflow and seems to work fine.

Reply:
oh, I see. Yeah that's weird that it overflows like that. 40 (or whatever the values from the trackpoint) * 5 should not overflow a 32 bit value even if it's signed.

Reply:
My cursor teleported to right bottom corner. Values sent to relative cursor update were somewhere around 8,589,934 which matches part of 2^33. Input seems to stay in rational range though.